static/app/views/performance/transactionSummary/transactionEvents/utils.tsx

```diff
@@ -1,4 +1,96 @@
-import {Query} from 'history';
+import {Location, Query} from 'history';
+
+import {t} from 'app/locale';
+import {decodeScalar} from 'app/utils/queryString';
+
+import {filterToField, SpanOperationBreakdownFilter} from '../filter';
+import {TransactionFilterOptions} from '../utils';
+
+export enum EventsDisplayFilterName {
+  p50 = 'p50',
+  p75 = 'p75',
+  p95 = 'p95',
+  p99 = 'p99',
+  p100 = 'p100',
+  avg_transaction_duration = 'avg_transaction_duration',
+}
+
+export type EventsDisplayFilter = {
+  name: EventsDisplayFilterName;
+  sort?: {kind: 'desc' | 'asc'; field: string};
+  label: string;
+  query?: string[][];
+};
+
+export type EventsFilterOptions = {
+  [name in EventsDisplayFilterName]: EventsDisplayFilter;
+};
+
+export type EventsFilterPercentileValues = {
+  [name in Exclude<EventsDisplayFilterName, EventsDisplayFilterName.p100>]: number;
+};
+
+export function getEventsFilterOptions(
+  spanOperationBreakdownFilter: SpanOperationBreakdownFilter,
+  percentileValues?: EventsFilterPercentileValues
+): EventsFilterOptions {
+  const {p99, p95, p75, p50, avg_transaction_duration} = percentileValues
+    ? percentileValues
+    : {p99: 0, p95: 0, p75: 0, p50: 0, avg_transaction_duration: 0};
+  return {
+    [EventsDisplayFilterName.p50]: {
+      name: EventsDisplayFilterName.p50,
+      query: p50 ? [['transaction.duration', `<=${p50.toFixed(0)}`]] : undefined,
+      sort: {
+        kind: 'desc',
+        field: filterToField(spanOperationBreakdownFilter) || 'transaction.duration',
+      },
+      label: t('p50'),
+    },
+    [EventsDisplayFilterName.p75]: {
+      name: EventsDisplayFilterName.p75,
+      query: p75 ? [['transaction.duration', `<=${p75.toFixed(0)}`]] : undefined,
+      sort: {
+        kind: 'desc',
+        field: filterToField(spanOperationBreakdownFilter) || 'transaction.duration',
+      },
+      label: t('p75'),
+    },
+    [EventsDisplayFilterName.p95]: {
+      name: EventsDisplayFilterName.p95,
+      query: p95 ? [['transaction.duration', `<=${p95.toFixed(0)}`]] : undefined,
+      sort: {
+        kind: 'desc',
+        field: filterToField(spanOperationBreakdownFilter) || 'transaction.duration',
+      },
+      label: t('p95'),
+    },
+    [EventsDisplayFilterName.p99]: {
+      name: EventsDisplayFilterName.p99,
+      query: p99 ? [['transaction.duration', `<=${p99.toFixed(0)}`]] : undefined,
+      sort: {
+        kind: 'desc',
+        field: filterToField(spanOperationBreakdownFilter) || 'transaction.duration',
+      },
+      label: t('p99'),
+    },
+    [EventsDisplayFilterName.p100]: {
+      name: EventsDisplayFilterName.p100,
+      label: t('p100'),
```

Comment:
Does p100 need to sorted by descending?

Reply:
I've left out `sort` from `p100` since I felt it was a slightly better experience for the user to display all events without replacing their existing sort when selecting `p100` from the dropdown. Let me know if you disagree with this, we can add the sort in!